app/controllers/vm_common.rb

```diff
@@ -573,15 +585,9 @@ def snap_vm
       @name = params[:name]
       @description = params[:description]
       if params[:name].blank? && !@record.try(:snapshot_name_optional?)
-        add_flash(_("Name is required"), :error)
-        @in_a_form = true
-        drop_breadcrumb(:name => _("Snapshot VM '%{name}'") % {:name => @record.name}, :url => "/vm_common/snap")
-        if session[:edit] && session[:edit][:explorer]
-          @edit = session[:edit]    # saving it to use in next transaction
-          javascript_flash(:spinner_off => true)
-        else
-          render :action => "snap"
-        end
+        render_missing_field(session, "Name")
+      elsif params[:description].blank? && @record.try(:snapshot_description_required?)
```

Comment:
It seems to be the opposite of what i want here,
http://api.rubyonrails.org/classes/Object.html#method-i-try-21
"try! - Same as try, but raises a NoMethodError exception if the receiver is not nil and does not implement the tried method."
All other providers do not currently implement this method so I just want it to return nil rather then throw an exception.